Codecademy/gamut
# feat(RadialProgress): updated with option to render outline around progress indicator

### Overview


Updated RadialProgress component with prop that adds an outline to the progress indicator and prop that adjusts SVG viewbox based on provided stroke-width


### PR Checklist

- [ ] Related to designs: https://www.figma.com/file/Y2xXPfZRbJco3QW0dqsFLx/Module-(Course)-Syllabus-Page?type=design&node-id=196-7717&mode=design&t=8FYMqnPlLeOVWkuZ-0
- [ ] Related to JIRA ticket: [EN-2344](https://codecademy.atlassian.net/browse/EN-2344)

#### Testing Instructions


**Monorepo:** See testing instructions on https://github.com/codecademy-engineering/mono/pull/2885

**Storybook:**
- Go to the `RadialProgress` component and see that there are new `progressOutline` and `adjustForStrokeWidth` props available to configure. 
- Check that the component renders with or without these props and doesn't produce any errors in the console. 
- Test with a `progressOutline`, set the prop to an object of the shape `{"size": 8, "color": "red"}` and see that it renders the outline around the "progress" portion of the component. 
- Test toggling the `adjustForStrokeWidth` prop on and off. While it's off (`false`), any stroke-width above `10` will be cut off by the viewport of the SVG. This is the existing behavior (prior to these changes) and will also happen if the `strokeWidth` + `progressOutline.size` is greater than `10` when added together. However when the `adjustForStrokeWidth` prop is toggled on (`true`), it will adjust for stroke-widths greater than `10` and will ensure the entire progress circle fits within the SVG viewport without being cut off.
- Test adding a `baseColor` prop with a custom color. When this is added, it should render the "non-progress" part of the circle in the provided custom color (in a solid/"1" opacity). When this prop is _not_ present, it will work as it did prior to these changes and will render the "non-progress" part of the circle in the provided `color` prop value but with "0.2" opacity.
<img width="1132" alt="image" src="https://github.com/Codecademy/gamut/assets/26662906/8f3c709e-bc5e-405b-a126-12b0d2483ce7">



#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/36215)  | 
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/2885)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2885)   |




[EN-2344]: https://codecademy.atlassian.net/browse/EN-2344?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
